controller/controller.go

```diff
@@ -282,7 +287,23 @@ func (c *Controller) bootWithError(ctx context.Context) error {
 	// users know when to go ahead. Note that mgr.Start below blocks the boot
 	// process until it ends gracefully or fails.
 	{
-		err = ctrl.Watch(&source.Kind{Type: c.newRuntimeObjectFunc()}, &handler.EnqueueRequestForObject{})
+		err = ctrl.Watch(
+			&source.Kind{Type: c.newRuntimeObjectFunc()},
+			&handler.EnqueueRequestForObject{},
+			predicate.Funcs{
```

Comment:
This makes sense but I've never seen `predicate.Funcs` before. Could you link to the upstream docs?

@xh3b4sd WDYT?

Reply:
It's here, https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/predicate . 
Also gitlab usage, https://gitlab.com/charts/components/gitlab-operator/commit/d55909edc4dbd47edef8deb0b1a980fc4b829d0a